### ISSUE TITLE
[MJAVADOC-672] misleading Exit code: 2 - javadoc: error - No source files for package

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#1053]
**[Ernst Reissner](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ernstreissner)** opened **[MJAVADOC-672](https://issues.apache.org/jira/browse/MJAVADOC-672?redirect=false)** and commented

Configuration is done as follows:

```java
// code 

      <!-- javadoc plugin -->
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-javadoc-plugin</artifactId>
        <version>3.1.1</version>
        <configuration>	  <!--phase>generate-sources</phase--> <!-- this is the default -->
	  <sourcepath>src/main/java:src/test/java</sourcepath>
	  <tags>
	    <tag>
	      <name>todo</name>
	      <placement>a</placement>
	      <head>what is to be done in the present code fragment;
	      jopas only</head>
	    </tag>
	    <tag>
	      <name>To.Do</name>
	      <placement>a</placement>
	      <head>ToDo: </head>
	    </tag>
	  </tags>

...

 </plugin>
```

The outcoming when doing mvn javadoc:test-javadoc is as follows:
ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.1.1:test-javadoc (default-cli) on project l2r: An error has occurred in Test Javadoc report generation:
[ERROR] Exit code: 2 - javadoc: error - No source files for package jopas
[ERROR]
[ERROR] Command line was: /usr/lib64/jvm/java/bin/javadoc `@options` `@argfile`
[ERROR]
[ERROR] Refer to the generated Javadoc files in '/home/ernst/Software/target/site/testapidocs' dir.

Note that the package is not org.jopas, but just jopas.
In the tag definition, there is a line break just before jopas.
If I remove that line break, all works fine.

I think this gives indication to very gragilepom reading.. either this plugin or maven.

---

**Affects:** 3.1.1

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Please share the options and args file. The issue is caused by the line break in `<head>`. Try to modify the files and replacing the physical line break with `\n` as literal. If that works we can evaluate options.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [f3c6b4a2582b5c689af9e8fabaea0d647e3ce9c4](https://gitbox.apache.org/repos/asf?p=maven-javadoc-plugin.git;a=commit;h=f3c6b4a2582b5c689af9e8fabaea0d647e3ce9c4)
